src/reporters/BroadcastReporter.ts

```diff
@@ -11,25 +10,17 @@ export interface NamedReporter {
   reporter: Reporter;
 }
 
-export const ALL_EVENT_METHOD_NAMES = ['onSourceFileRead', 'onAllSourceFilesRead', 'onAllMutantsMatchedWithTests', 'onMutantTested', 'onAllMutantsTested', 'onConfigRead'];
-
-export default class BroadcastReporter implements Reporter {
+export default class BroadcastReporter implements StrictReporter {
 
   constructor(private reporters: NamedReporter[]) {
-    ALL_EVENT_METHOD_NAMES.concat('wrapUp').forEach(method => {
-      (<any>this)[method] = (arg: any) => {
-        return this.broadcast(method, arg);
-      };
-    });
   }
 
-  private broadcast(methodName: string, eventArgs: any) {
+  private broadcast(methodName: keyof Reporter, eventArgs: any = undefined): Promise<any> | void {
```

Comment:
I use the `keyof Reporter` union type to force a string is of type `'onSourceFileRead' | 'onAllSourceFilesRead' | 'onAllMutantsMatchedWithTests' | 'onMutantTested' | 'onAllMutantsTested' | 'wrapUp'`